ocaml/xapi-storage/generator/lib/control.ml

```diff
@@ -421,9 +421,15 @@ module Volume (R : RPC) = struct
   let compose =
     R.declare "compose"
       [
-        "[compose sr volume1 volume2] layers the updates from [volume2] onto"
-      ; "[volume1], modifying [volume2]. Implementations shall declare the"
-      ; "VDI_COMPOSE feature for this method to be supported."
+        "[compose sr child_volume parent_volume] layers the updates from "
+      ; "[child_volume] onto [parent_volume], modifying [child_volume]. "
```

Comment:
The old text says `modifying [volume2]`, which is the second volume parameter, but the `child_volume` is the first one. Did the old text have the params the wrong way around?

Reply:
either that or the implementation did, it was very unclear from the original doc comment what the expectation was with regards to the actual usage. Now that we have an implementation this PR makes things consistent.

Reply:
I think also it's difficult to say what gets modified.

In the case of a delta file format like vhd or qcow2 it is the child that is modified to have a parent pointer and the parent becomes a hidden entity that is no longer directly accessible (this is analogous to the behaviour of vdi_compose in SMAPIv1).

For alternative implementations such as reflinks or object based storage it may be the parent which is modified by having the data for the child merged into it and then being renamed to the child's id. As we do not currently have any implementations working in this fashion this is something that will need to be resolved when we do.